src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_final.rs

```diff
@@ -136,28 +136,19 @@ where Method: HashMethodBounds
                         let mut states_binary_columns = Vec::with_capacity(states_columns.len());
 
                         for agg in states_columns.iter().take(aggregate_function_len) {
-                            let aggr_column =
-                                agg.value.as_column().unwrap().as_string().ok_or_else(|| {
-                                    ErrorCode::IllegalDataType(format!(
-                                        "Aggregation column should be StringType, but got {:?}",
-                                        agg.value
-                                    ))
-                                })?;
-                            states_binary_columns.push(aggr_column);
+                            let col = agg.value.as_column().unwrap();
+                            states_binary_columns.push(col.slice(0..places.len()));
```

Comment:
`places.len()` should equal to `agg.len()`. We can add an assertion and remove the slice operation.

Reply:
there is `reach_limit` which could make places size shorter than `agg.len()`